main.go

```diff
@@ -170,6 +170,10 @@ func runTaskLoop(ctx context.Context, c chan struct{}) {
 		// Set debug logging settings so that customers don't need to restart the agent.
 		logger.SetDebugLogging(agentconfig.Debug())
 		clog.DebugEnabled = agentconfig.Debug()
+		if agentconfig.TaskNotificationEnabled() && taskNotificationClient == nil {
+			// Call RegisterAgent now since we just we either just started running or were just enabled.
+			registerAgent(ctx)
```

Comment:
Not blocking this PR, but we should also call RegisterInventory here. Something to track

Reply:
Inventory happens on it's own schedule (default once every 10 min) so between agent being enabled and an inventory run will be <10min which is totally fine.